File: dataset_reduzido/codenet/Java/Code/s973587235.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        // 整数の入力
        int n = sc.nextInt();
        int minN = sc.nextInt()-1;
        for(int i=0;i<n-1;i++){
            int h = sc.nextInt();
            if( minN == h){
                minN = h;
            }else if(minN == h-1){
                minN = h-1;
            }else if(minN < h-1){
                minN = h-1;
            }else if(minN > h){
                System.out.println("No");
                return;
            }
        }
        System.out.println("Yes");
    }
}

File: output/Llama-2-100/codenet/Java/Python/s973587235.py
import sys

def main():
    sc = input()
    n = int(sc)
    min_n = int(sc) - 1
    for i in range(n - 1):
        h = int(sc)
        if min_n == h:
            min_n = h
        elif min_n == h - 1:
            min_n = h - 1
        elif min_n < h - 1:
            min_n = h - 1
        elif min_n > h:
            print("No")
            return
    print("Yes")

